File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.java
import java.util.ArrayList;

class Solution {
        public int solution(int n, int w, int num) {
        int answer = 0;
        boolean direction = true;
        ArrayList<ArrayList<Integer>> board = new ArrayList<>();


        for (int i = 0; i < w; i++) {
            board.add(new ArrayList<Integer>());
        }

        for (int i = 0; i < n; i++) {
            if(i % w == 0 && i != 1 && i != 0){
                direction = !direction;
            }

            if (direction) {
                board.get(i % w).add(i + 1);
            } else {
                board.get(w - 1 - i % w).add(i + 1);
            }
        }
   
        for(ArrayList<Integer> arr : board){
            if(arr.contains(num)){
                int idx = arr.indexOf(num);
                answer = arr.subList(idx,arr.size()).size();
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    answer = 0
    # 각 board에는 같은 행에 속하는 숫자들이 속하게하기 위해 w만큼 리스트 추가
    board = [[] for _ in range(w)]
    direction = True
    for i in range(n):
        if i % w == 0 and i != 1 and i != 0:
            direction = not direction
        if direction:
            board[i%w].append(i+1)
        else:
            board[w - 1 - i%w].append(i+1)


    for b in board:
        if num in b:
            idx = b.index(num)
            answer = len(b[idx:])

    return answer